### PR TITLE
Add Timeline Block

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes https://github.com/Codeinwp/otter-internals/issues/177

### Summary

Added the initial implementation for Timeline Block

### Screenshots 

image

image

https://github.com/user-attachments/assets/ea9c5828-4448-451d-b735-857c58389977

https://github.com/user-attachments/assets/1ee108ff-7d21-41e2-87fa-b632f63b4686

----

### Test instructions

1. Inser the Timeline Block
2. Customize 
3. Check if everything is working

---- 

### Checklist before the final review

- [x] Included E2E or unit tests for the changes in this PR.
- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

**Comment by ineagu** [comment]
Tried it out with the default theme on instawp, is failing me : https://thoughtful-chipmunk-87d693.vertisite.cloud/our-history/  https://vertis.d.pr/i/KrJXfR. 

It is supposed to look like that?

**Comment by Soare-Robert-Daniel** [comment]
@ineagu, is now fixed. I enforced some styling that was missing on instances like TasteWP

**Comment by ineagu** [comment]
@Soare-Robert-Daniel, that won't work for me. It works for Spectra because you can't add blocks within the timeline items. For us, I see "+" signs everywhere, and I usually go to the top to select the parent: https://vertis.d.pr/i/QzjWDq. I can't see the "+" that you add at the bottom.

I suggest we go with this approach: https://vertis.d.pr/i/GhcT7g. This is displayed both when you select the parent or child blocks and is not displayed if other blocks on the page are selected.

File: inc/class-blocks-css.php
Comment (HardeepAsrani) [inline inc/class-blocks-css.php:1]: Not sure why but your rebase is undoing some of the changes that were made in other PRs. This is one of those.

File: inc/class-registration.php [hunk 1]
```diff
@@ -279,11 +279,10 @@ public function enqueue_block_editor_assets() {
 				'showOnboarding'          => $this->show_onboarding(),
 				'ratingScale'             => get_option( 'themeisle_blocks_settings_review_scale', false ),
 				'hasModule'               => array(
-					'aiToolbar'       => boolval( get_option( 'themeisle_blocks_settings_block_ai_toolbar_module', true ) ),
 					'blockCSS'        => boolval( get_option( 'themeisle_blocks_settings_css_module', true ) ),
 					'blockAnimations' => boolval( get_option( 'themeisle_blocks_settings_blocks_animation', true ) ),
 					'blockConditions' => boolval( get_option( 'themeisle_blocks_settings_block_conditions', true ) ),
-					'patternsLibrary' => boolval( get_option( 'themeisle_blocks_settings_patterns_library', true ) ),
+					'aiToolbar'       => boolval( get_option( 'themeisle_blocks_settings_block_ai_toolbar_module', true ) ),
```
Comment (HardeepAsrani) [inline inc/class-registration.php:285]: This is also being reverted.

File: inc/plugins/class-block-conditions.php
Comment (HardeepAsrani) [inline inc/plugins/class-block-conditions.php:1]: This is one of many cases. Can you fix that before the review?

**Comment by Soare-Robert-Daniel** [comment]
I will remake this branch

**Pull request closed without merging**